File: app/src/main/java/com/oliver/weatherapp/data/remote/DayWeatherToWeatherEntryMapper.java
package com.oliver.weatherapp.data.remote;

import com.oliver.weatherapp.data.local.model.WeatherEntry;
import com.oliver.weatherapp.data.remote.model.DayWeather;

import java.util.Date;

public class DayWeatherToWeatherEntryMapper  {

    public WeatherEntry apply(DayWeather dayWeather) {
        WeatherEntry entry = new WeatherEntry();
        if (dayWeather.weather != null && !dayWeather.weather.isEmpty()) {
            entry.weatherIconId = dayWeather.weather.get(0).id;
        }
        entry.date = new Date(dayWeather.date * 1000);
        if (dayWeather.main != null) {
            entry.min = dayWeather.main.tempMin;
            entry.max = dayWeather.main.tempMax;
            entry.humidity = dayWeather.main.humidity;
        }
        if (dayWeather.wind != null) {
            entry.wind = dayWeather.wind.speed;
            entry.degrees = dayWeather.wind.deg;
        }

        if (dayWeather.rain != null) {
            entry.rainVolume = dayWeather.rain.rainVolume;
        }
        return entry;
    }
}

File: app/src/main/java/com/oliver/weatherapp/data/remote/DayWeatherToWeatherEntryMapper.kt
package com.oliver.weatherapp.data.remote

import com.oliver.weatherapp.data.local.model.WeatherEntry
import com.oliver.weatherapp.data.remote.model.DayWeather
import java.util.*

class DayWeatherToWeatherEntryMapper {

    fun apply(dayWeather: DayWeather): WeatherEntry {
        val entry = WeatherEntry()

        dayWeather.weather?.let {
            if (it.isNotEmpty())
                entry.weatherIconId = it[0].id
        }
        entry.date = Date(dayWeather.date * 1000)

        dayWeather.main?.let {
            entry.min = it.tempMin
            entry.max = it.tempMax
            entry.humidity = it.humidity.toDouble()
        }

        dayWeather.wind?.let {
            entry.wind = it.speed
            entry.degrees = it.deg
        }

        dayWeather.rain?.let {
            entry.rainVolume = it.rainVolume
        }
        return entry
    }
}
